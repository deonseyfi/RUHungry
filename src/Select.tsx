import React from 'react';
import { createStyles, makeStyles, Theme } from '@material-ui/core/styles';
import { Select as MuiSelect, FormControl, Input } from '@material-ui/core';

const useStyles = makeStyles((theme: Theme) =>
    createStyles({
        formControl: {
            margin: theme.spacing(1),
            width: '3rem',
            fontSize: '25',
            textAlign: 'center',
        },
        selectEmpty: {
            marginTop: theme.spacing(2),
        },
        select: {
            color: 'white',
        },
        icon: { color: 'white' },
        underline: {
            borderBottom: '2px solid white',
            '&:after': {
                borderBottom: '2px solid white',
            },
        },
    }),
);

/**
 * Native Select component.
 *
 * @returns {React.ReactElement} A custom Select.
 */
const Select = (): React.ReactElement => {
    const classes = useStyles();
    const [category, setCategory] = React.useState('');
    const handleChange = (event: React.ChangeEvent<{ value: unknown }>) => {
        setCategory(event.target.value as string);
    };
    return (
        <FormControl className={classes.formControl} style={{ fontSize: 50 }}>
            <MuiSelect
                native
                value={category}
                onChange={handleChange}
                inputProps={{
                    name: 'category',
                }}
                classes={{
                    select: classes.select,
                    icon: classes.icon,
                }}
                input={<Input classes={{ underline: classes.underline }} />}
            >
                <option aria-label='None' value='' disabled />
                <option value='smile'>😀</option>
                <option value='cat'>🐈</option>
                <option value='dog'>🐶</option>
            </MuiSelect>
        </FormControl>
    );
};

export default Select;
