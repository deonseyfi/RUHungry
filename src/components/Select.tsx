import React from 'react';
import { createStyles, makeStyles, Theme } from '@material-ui/core/styles';
import { Select as MuiSelect, FormControl, Input } from '@material-ui/core';

const useStyles = makeStyles((theme: Theme) =>
    createStyles({
        formControl: {
            margin: theme.spacing(1),
            width: 'auto',
            fontSize: '25',
            textAlign: 'center',
        },
        selectEmpty: {
            marginTop: theme.spacing(2),
        },
        select: { color: 'gray' },
        icon: { color: 'white' },
        underline: {
            borderBottom: '2px solid white',
            '&:after': {
                borderBottom: '2px solid white',
            },
        },
    }),
);

/**
 * Select props.
 */
interface SelectProps {
    /**
     * List of items in the Select drop-down.
     */
    children: {
        /**
         * Text representation of the drop-down child item.
         */
        value: string;
        /**
         * Emoji representation of the drop-down child item.
         */
        symbol: string;
    }[];
    /**
     * Value of the selected item.
     */
    value: string;
    /**
     * Handle a change of selection.
     */
    onChange: (event: React.ChangeEvent<{
        /**
         * Value being passed into the onChange call.
         */
        value: unknown;
    }>,) => void
}

/**
 * Select component.
 *
 * @param {SelectProps} props Select props.
 * @returns {React.ReactElement<SelectProps>} Select component with props.
 */
const Select = (props: SelectProps): React.ReactElement<SelectProps> => {
    const classes = useStyles();
    // Map list of children into HTML Option items for Select
    const list = props.children.map((items) =>
        <option key={items.value} value={items.value}>{items.symbol}</option>);

    return (
        <FormControl className={classes.formControl}>
            <MuiSelect
                native
                value={props.value}
                onChange={props.onChange}
                // inputProps={{
                //     style: { fontSize: 16 },
                //     name: 'value',
                // }}
                classes={{
                    select: classes.select,
                    icon: classes.icon,
                }}
                input={<Input classes={{ underline: classes.underline }} />}
            >
                {list}
            </MuiSelect>
        </FormControl>
    );
};

export default Select;
