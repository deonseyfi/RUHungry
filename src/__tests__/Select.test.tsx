import React from 'react';
import { mount, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Select from '../components/Select';

configure({ adapter: new Adapter() });

const list = [{ value: 'All', symbol: 'All' }];
const value = 'All';
const onChange = jest.fn();

describe('Select test', () => {
    test('renders Select component', () => {
        const select = mount(
            <Select value={value} onChange={onChange}>
                {list}
            </Select>,
        );
        expect(select).toMatchSnapshot();
    });
});
