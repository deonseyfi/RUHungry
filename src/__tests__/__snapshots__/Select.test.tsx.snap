// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Select test renders Select component 1`] = `ReactWrapper {}`;
